hospital/src/main/java/by/emel/anton/model/dao/implementation/FileDoctorDAO.java

```diff
@@ -10,26 +10,29 @@
 
 import java.io.*;
 import java.time.LocalDate;
-import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 
 public class FileDoctorDAO implements DoctorDAO {
 
     @Override
-    public Doctor getDoctor(String login, String password) throws UserDAOException {
+    public Optional<Doctor> getDoctor(String login, String password) throws UserDAOException {
         try(BufferedReader bufferedReader = new BufferedReader(new FileReader(Constans.FILE_PATH_DOCTORS))) {
```

Comment:
This is old-fashioned way of reading file. You can make use of `Files` utility helper and stream all lines of a file